Cannot tsc on 5.12.0

### Description

We cannot build with the latest version.
tsconfig.json:
```json
{
    "compilerOptions": {
        "module": "node18",
        "target": "es2022", 
        "strict": true,
        "verbatimModuleSyntax": true,
        "declaration": true,
        "sourceMap": true,
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "resolveJsonModule": true,
        "declarationMap": true,
        "rootDirs": ["src"],
        "outDir": "build",
        "types": ["vitest/globals"]
    },
    "include": ["src/**/*", "test/**/*"],
    "exclude": ["node_modules", "tsconfig.json"]
}
```

### Node.js Version

22.16

### Redis Server Version

_No response_

### Node Redis Version

5.12.0

### Platform

Linux

### Logs

```bash
node_modules/@redis/client/dist/lib/opentelemetry/metrics.d.ts:7:28 - error TS2307: Cannot find module '@opentelemetry/api' or its corresponding type declarations.

7         api: typeof import("@opentelemetry/api");
                             ~~~~~~~~~~~~~~~~~~~~

node_modules/@redis/client/dist/lib/opentelemetry/types.d.ts:1:95 - error TS2307: Cannot find module '@opentelemetry/api' or its corresponding type declarations.

1 import { Attributes, Counter, Histogram, MeterProvider, ObservableGauge, UpDownCounter } from "@opentelemetry/api";
                                                                                                ~~~~~~~~~~~~~~~~~~~~


Found 2 errors in 2 files.

Errors  Files
     1  node_modules/@redis/client/dist/lib/opentelemetry/metrics.d.ts:7
     1  node_modules/@redis/client/dist/lib/opentelemetry/types.d.ts:1
```

--- a/packages/client/lib/opentelemetry/index.ts
+++ b/packages/client/lib/opentelemetry/index.ts
@@ -1,7 +1,7 @@
 import { OpenTelemetryError } from "../errors";
 import { ClientRegistry } from "./client-registry";
 import { OTelMetrics } from "./metrics";
-import { ObservabilityConfig } from "./types";
+import { ObservabilityConfig, OpenTelemetryApiModule } from "./types";
 
 export class OpenTelemetry {
   private static _instance: OpenTelemetry | null = null;
@@ -57,7 +57,7 @@ export class OpenTelemetry {
       throw new OpenTelemetryError("OpenTelemetry already initialized");
     }
 
-    const api: typeof import("@opentelemetry/api") = (() => {
+    const api: OpenTelemetryApiModule = (() => {
       try {
         return require("@opentelemetry/api");
       } catch {

--- a/packages/client/lib/opentelemetry/metrics.ts
+++ b/packages/client/lib/opentelemetry/metrics.ts
@@ -1,10 +1,12 @@
 import type * as DC from 'node:diagnostics_channel';
-import { Meter, BatchObservableResult } from "@opentelemetry/api";
 import { ClientRegistry } from "./client-registry";
 import {
+  BatchObservableResult,
   DEFAULT_OTEL_ATTRIBUTES,
+  Meter,
   MetricInstruments,
   ObservabilityConfig,
+  OpenTelemetryApiModule,
   OTEL_ATTRIBUTES,
   MetricOptions,
   DEFAULT_METRIC_GROUPS,
@@ -499,7 +501,7 @@ export class OTelMetrics {
   readonly #options: MetricOptions;
 
   private constructor(
-    api: typeof import("@opentelemetry/api"),
+    api: OpenTelemetryApiModule,
     config?: ObservabilityConfig,
   ) {
     this.#options = this.parseOptions(config);
@@ -535,7 +537,7 @@ export class OTelMetrics {
     api,
     config,
   }: {
-    api: typeof import("@opentelemetry/api");
+    api: OpenTelemetryApiModule;
     config?: ObservabilityConfig;
   }) {
     if (OTelMetrics.#initialized) {
@@ -568,7 +570,7 @@ export class OTelMetrics {
 
 
   #getMeter(
-    api: typeof import("@opentelemetry/api"),
+    api: OpenTelemetryApiModule,
     options: MetricOptions,
   ): Meter {
     if (options.meterProvider) {

--- a/packages/client/lib/opentelemetry/types.ts
+++ b/packages/client/lib/opentelemetry/types.ts
@@ -1,12 +1,79 @@
-import {
-  Attributes,
-  Counter,
-  Histogram,
-  MeterProvider,
-  ObservableGauge,
-  UpDownCounter,
-} from "@opentelemetry/api";
 import { version } from "../../package.json";
+
+type AttributePrimitive = string | number | boolean;
+
+// Keep the public OTel-facing config structurally typed so consumers do not
+// need @opentelemetry/api installed unless they actually initialize it.
+export type AttributeValue =
+  | AttributePrimitive
+  | readonly AttributePrimitive[];
+
+export type Attributes = Record<string, AttributeValue | undefined>;
+
+type InstrumentOptions = {
+  unit?: string;
+  description?: string;
+};
+
+export interface Counter<TAttributes extends Attributes = Attributes> {
+  add(value: number, attributes?: TAttributes, context?: unknown): void;
+}
+
+export interface Histogram<TAttributes extends Attributes = Attributes> {
+  record(value: number, attributes?: TAttributes, context?: unknown): void;
+}
+
+export interface UpDownCounter<TAttributes extends Attributes = Attributes> {
+  add(value: number, attributes?: TAttributes, context?: unknown): void;
+}
+
+export interface ObservableGauge<TAttributes extends Attributes = Attributes> {
+  readonly _attributes?: TAttributes;
+}
+
+export interface BatchObservableResult {
+  observe<TAttributes extends Attributes>(
+    instrument: ObservableGauge<TAttributes>,
+    value: number,
+    attributes?: TAttributes,
+    context?: unknown,
+  ): void;
+}
+
+export interface Meter {
+  createHistogram(
+    name: string,
+    options?: InstrumentOptions & {
+      advice?: {
+        explicitBucketBoundaries?: number[];
+      };
+    },
+  ): Histogram<Attributes>;
+  createCounter(
+    name: string,
+    options?: InstrumentOptions,
+  ): Counter<Attributes>;
+  createUpDownCounter(
+    name: string,
+    options?: InstrumentOptions,
+  ): UpDownCounter<Attributes>;
+  createObservableGauge(
+    name: string,
+    options?: InstrumentOptions,
+  ): ObservableGauge<Attributes>;
+  addBatchObservableCallback(
+    callback: (observableResult: BatchObservableResult) => void,
+    observables: ObservableGauge[],
+  ): void;
+}
+
+export interface MeterProvider {
+  getMeter(name: string, version?: string, options?: unknown): Meter;
+}
+
+export interface OpenTelemetryApiModule {
+  metrics: Pick<MeterProvider, "getMeter">;
+}
 
 export const METRIC_GROUP = {
   COMMAND: "command",

